R/rw.R

```diff
@@ -0,0 +1,251 @@
+  class(.rw) <- append(class(.rw), 'rw_norm')
+  return (.rw)
+}
+
+#' @export
+rw_unit <- function (.rw = rw(), conf.index) {
+  stopifnot(inherits(.rw, 'rw'))
+
+  .rw$conf.index <- conf.index
+  .rw$rw <- rep(1.0,length(conf.index))
+  .rw$max_value <- 1.0
+
+  class(.rw) <- append(class(.rw), 'rw_orig')
+  class(.rw) <- append(class(.rw), 'rw_meta')
+  class(.rw) <- append(class(.rw), 'rw_norm')
+  return (.rw)
+}
+
+#' Checks whether the cf object contains no data
+#'
+#' @param .rw `rw` object.
+#'
+#' @examples
+#' # The empty rw object must be empty:
+#' is_empty.rw(rw())
+#'
+is_empty.rw <- function (.rw) {
+  setequal(class(.rw), class(rw())) &&
+    is.null(names(.rw))
+}
+
+
+#' Arithmetically multiplies two reweighting factors 
+#'
+#' @param rw1,rw2 `rw_orig` objects to be muplitplied
+#' @param nf1,nf2 Integer. Factors that determines the number of flavours we
+#'        have for reweighting factor 1 and 2, the default value is 1, because
+#'        usually we compute the reweighting factors for Q: the product of 
+#'        up and down or strange and charm determinant, there are no additional 
+#'        terms in the sea determinant that have to be reweighted with the same
+#'        rw factor. In case we compute the rw1 factor only for the up quark, we 
+#'        for example, we have to set nf1=2 to obtain the rw factor
+#'        for the light determinant.  
+#'
+#' @return
+#' The value is
+#' \deqn{rw_1*nf1*rw_2*nf2) \,.}
+#'
+#' @export
+multiply.rw <- function(rw1, rw2, nf1 = 1, nf2 = 1) {
+
+  stopifnot(inherits(rw1, 'rw'))
+  stopifnot(inherits(rw2, 'rw'))
+ 
+  stopifnot(inherits(rw1, 'rw_orig'))
+  stopifnot(inherits(rw2, 'rw_orig'))
+
+  stopifnot(inherits(rw1, 'rw_meta'))
+  stopifnot(inherits(rw2, 'rw_meta'))
+ 
+  stopifnot(identical( rw1$conf.index, rw2$conf.index ))
+
+  rw <- rw()
+  rw$rw <- nf1*rw1$rw * nf2*rw2$rw
+  rw$max_value <- NA
+  rw$conf.index <- rw1$conf.index 
+
+  class(rw) <- append(class(rw), 'rw_orig')
+  class(rw) <- append(class(rw), 'rw_meta')
+  
+  return(rw)
+}
+
+#' Arithmetically multiply reweighting factors
+#'
+#' @param rw1,rw2 `rw_orig` objects.
+#'
+#' @export
+'*.rw' <- function (rw1, rw2) {
+  multiply.rw(cf1, cf2, nf1 = 1, nf2 = 1)
+}
+
+
+#' Checks whether an object is a rw
+#'
+#' @param x Object, possibly of class `rw`.
+#'
+#' @export
+is.rw <- function (x) {
+  inherits(x, "rw")
+}
+
+meanval <- function(d, w) {return(mean(d$rw[w]))}
+
+#' Plot a reweighting factor function
+#'
+#' @param x `rw` object
+#' @param rep See \code{\link{plotwitherror}}.
+#'
+#' @inheritParams plotwitherror
+#'
+#' @export
+plot.rw <- function(x, rep = FALSE, ...) {
+  rw <- x
+  stopifnot(inherits(rw, 'rw'))
+  negs <- which(rw$rw < 0)
+  neg.vec <- rep(1,times=length(rw$rw))
+  neg.vec[negs] <- -1
+ 
+  colneg <- rep("black",times=length(rw$rw))
+  colneg[negs] <- "red"
+
+# Estimating the statistical error on the reweighting factor
```

Comment:
I don't understand. This error should be the from the stochastic samples for the reweighting factor.